### PR TITLE
Update Dependencies for Stability

### PR DESCRIPTION
**Pull request by MostlyKIGuess**
## Description:
This PR updates the development dependencies in the sugar-web project to resolve potential vulnerabilities and improve compatibility with modern tooling. The following changes have been made:

### Key Updates
1. **Dependencies Updated**:
   - `grunt`: `^1.6.1`
   - `grunt-karma`: `^4.0.2`
   - `requirejs`: `^2.3.7`
   - `karma`: `^6.4.4`
   
   2. **Validation Performed**:
   - Verified `grunt` tasks (`jsbeautifier`, `jshint`, `test`) run without errors.
   - Ran tests using Karma (`test/karma-unit.conf.js`) to ensure all tests pass.
   - Removed and reinstalled `node_modules` to confirm clean installation.

### Testing
- All tasks executed successfully using the following commands:

  ```bash
  grunt
  npx karma start test/karma-unit.conf.js
  npx karma start test/karma-functional.conf.js 

### Checklist

- [x] Updated dependencies in package.json.
- [x] Verified compatibility using grunt and karma.
- [x] Ensured clean installation and execution of all tasks.


![Grunt](https://github.com/user-attachments/assets/0a22177c-d055-4f25-827a-0e3f78774ba2)

![Functional Test](https://github.com/user-attachments/assets/39bc3dca-def5-4f6a-aa8f-7c7570857d66)

![Unit Test](https://github.com/user-attachments/assets/19649c8b-f9b0-4f89-93d6-0c94111a5356)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /lib/
 /node_modules/
+package-lock.json
```
Comment (quozl) [inline .gitignore:3]: Missing newline at end of file.

File: Gruntfile.js [hunk 2]
```diff
@@ -13,12 +13,12 @@ module.exports = function (grunt) {
         karma: {
             unit: {
                 configFile: 'test/karma-unit.conf.js',
-                browsers: ['Chrome'],
+                browsers: [ 'Firefox'],
```
Comment (quozl) [inline Gruntfile.js:16]: Space before quote.

File: test/functional/envSpec.js [hunk 3]
```diff
@@ -39,59 +32,48 @@ define(["sugar-web/env"], function (env) {
...
                     window.top.sugar.environment = expectedEnv;
                     window.top.sugar.onEnvironmentSet();
-
-                    expect(callback).toHaveBeenCalledWith(null, expectedEnv);
                 });
             });
         });
     });
-});
+});
```
Comment (quozl) [inline test/functional/envSpec.js:79]: Missing newline at end of file.

File: test/unit/busSpec.js [hunk 4]
```diff
@@ -121,36 +92,20 @@ define(["sugar-web/bus"], function (bus) {
...
-            runs(function () {
-                expect(notificationParams).toEqual(originalParams);
-            });
+            bus.onNotification("hey.there", onNotificationReceived);
+            client.params = originalParams;
+            client.send_notification("hey.there");
         });
     });
-});
+});
```
Comment (quozl) [inline test/unit/busSpec.js:111]: Missing newline at end of file.

File: test/unit/datastoreSpec.js [hunk 5]
```diff
@@ -1,32 +1,27 @@
...
+                expect(datastoreObject.objectId).toBeDefined();
+                expect(datastoreObject.objectId).toEqual(objectId);
+                done();
             });
+
+            callback();
         });
     });
-});
+});
```
Comment (quozl) [inline test/unit/datastoreSpec.js:27]: Missing newline at end of file.

File: test/unit/dictstoreSpec.js [hunk 6]
```diff
@@ -1,54 +1,40 @@
...
+                dictstore.save(function () {
                     expect(localStorage.test_key).toBe("test");
+                    done();
                 });
             });
         });
-
     });
-});
+});
```
Comment (quozl) [inline test/unit/dictstoreSpec.js:40]: Missing newline at end of file.

File: test/unit/envSpec.js [hunk 7]
```diff
@@ -1,46 +1,33 @@
...
-            }, "environment not to be undefined");
-
-            runs(function () {
-                expect(actualEnv).toEqual({});
-            });
-
         });
     });
-});
+});
```
Comment (quozl) [inline test/unit/envSpec.js:33]: Missing newline at end of file.

File: test/unit/graphics/iconSpec.js [hunk 8]
```diff
@@ -1,59 +1,35 @@
...
+                    expect(secondUrl).not.toBe(firstIconUrl);
+                    done();
+                });
             });
-
         });
     });
-
-});
+});
```
Comment (quozl) [inline test/unit/graphics/iconSpec.js:35]: Missing newline at end of file.

File: test/unit/graphics/menupaletteSpec.js [hunk 9]
```diff
@@ -73,6 +54,5 @@ define(["sugar-web/graphics/menupalette", "sugar-web/graphics/palette"], functio
         it("should be an instance of the parent class", function () {
             expect(menuPalette instanceof palette.Palette).toBe(true);
         });
-
     });
-});
+});
```
Comment (quozl) [inline test/unit/graphics/menupaletteSpec.js:58]: Missing newline at end of file.

File: test/unit/graphics/paletteSpec.js [hunk 10]
```diff
@@ -30,7 +29,5 @@ define(["sugar-web/graphics/palette"], function (palette) {
             expect(myPaletteA.isDown()).toBe(true);
             expect(myPaletteB.isDown()).toBe(false);
         });
-
     });
-
-});
+});
```
Comment (quozl) [inline test/unit/graphics/paletteSpec.js:33]: Missing newline at end of file.

File: test/unit/graphics/radiobuttonsgroupSpec.js [hunk 11]
```diff
@@ -14,84 +12,48 @@ define(["sugar-web/graphics/radiobuttonsgroup"], function (
...
-                expect(elem).toBe(elem1);
-                expect(elem.classList.contains('active')).toBe(true);
-            });
+            elem2.click();
         });
-
     });
-
-});
+});
```
Comment (quozl) [inline test/unit/graphics/radiobuttonsgroupSpec.js:59]: Missing newline at end of file.

**Review by quozl — CHANGES_REQUESTED** [review]
Thanks.  Please;
* [ ] run a test of a Sugar activity with updated sugar-web, such as the [hello world](https://github.com/sugarlabs/hello-web-activity) activity,
* [ ] rewrite the commit messages according to [our guidance](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits).

File: .gitignore (hunk 1 above)
Comment (MostlyKIGuess) [reply .gitignore]: Updated all of the changes in the new commit.

**Comment by MostlyKIGuess** [comment]
> Thanks. Please;
> 
>     * [ ]  run a test of a Sugar activity with updated sugar-web, such as the [hello world](https://github.com/sugarlabs/hello-web-activity) activity,
> 
>     * [ ]  rewrite the commit messages according to [our guidance](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#m




-  [x]  run a test of a Sugar activity with updated sugar-web, such as the [hello world](https://github.com/sugarlabs/hello-web-activity) activity,
- Video of me doing the updated test:


https://github.com/user-attachments/assets/e7d7f551-3c4f-4b20-bd8b-4cc76507bb84


- [ ]  rewrite the commit messages according to [our guidance](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits).
- I am not sure how to do this, I was having trouble, and I did squash some commits but not sure if that worked 😅 .

**Comment by MostlyKIGuess** [comment]
I have also added new line..

**Comment by quozl** [comment]
Thanks, but you've simply run activities using file:// inside your own browser; for web activities Sugar runs an embedded browser and provides access to Journal through an API.  Please test inside Sugar.

A way to rewrite commit messages is `git rebase -i` and the `reword` command, then `git push --force`.

**Comment by MostlyKIGuess** [comment]
- Test for Connect-the-dots and Hello-World with the new sugar-web.
- cloned from my own forks with updated library as mentioned and tested on a virtual machine of sugar live distro.
https://github.com/user-attachments/assets/760e5857-ac45-4b94-896b-1a1324925000

**Comment by MostlyKIGuess** [comment]
@quozl should I repoen a PR with just one commit? Currently rebasing is causing conflicts.

**Comment by quozl** [comment]
No.  Do any rebasing or squashing in branches of your own repository clone, and when you are happy with the result reset your pull request branch to the git hash of your happy state and force push.

I note you have used your _master_ branch as pull request against our recommendation.  Yes, this is one of the reasons why we recommend against that.

**Pull request closed without merging**